File: iOS/DailyWater/Models/DailyRecord.swift
import Foundation

/// Archived summary for one completed day of water tracking.
struct DailyRecord: Codable, Identifiable, Equatable {
    var id: String { dateKey }
    let dateKey: String      // yyyy-MM-dd
    let totalIntake: Double
    let goal: Double
    let unit: String

    /// Progress ratio clamped to [0, 1].
    var progress: Double {
        guard goal > 0 else { return 0 }
        return min(totalIntake / goal, 1.0)
    }

    /// Whether the goal was met on this day.
    var goalMet: Bool {
        goal > 0 && totalIntake >= goal
    }

    /// Parse dateKey back into a Date (start of day, local TZ).
    var date: Date? {
        let formatter = DateFormatter()
        formatter.dateFormat = "yyyy-MM-dd"
        formatter.calendar = Calendar.current
        formatter.timeZone = TimeZone.current
        return formatter.date(from: dateKey)
    }
}

File: Android/app/src/main/java/com/watercamel/model/DailyRecord.kt
package com.watercamel.model

/**
 * Archived summary for one completed day of water tracking.
 */
data class DailyRecord(
    val dateKey: String,       // yyyy-MM-dd
    val totalIntake: Double,
    val goal: Double,
    val unit: String
) {
    /** Progress ratio clamped to [0, 1]. */
    val progress: Double
        get() = if (goal > 0) minOf(totalIntake / goal, 1.0) else 0.0

    /** Whether the goal was met on this day. */
    val goalMet: Boolean
        get() = goal > 0 && totalIntake >= goal
}
